nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/flow-designer/state/controller-services/controller-services.effects.ts

```diff
@@ -436,9 +514,24 @@ export class ControllerServicesEffects {
             this.actions$.pipe(
                 ofType(ControllerServicesActions.configureControllerServiceSuccess),
                 map((action) => action.response),
-                tap((response) => {
-                    if (response.postUpdateNavigation) {
-                        this.router.navigate(response.postUpdateNavigation);
+                concatLatestFrom(() => this.store.select(selectCurrentProcessGroupId)),
+                tap(([response, processGroupId]) => {
+                    if (response.postUpdateNavigation && response.postUpdateNavigationBoundary) {
+                        this.router.navigate(response.postUpdateNavigation, {
+                            state: {
+                                backNavigation: {
+                                    route: [
+                                        '/process-groups',
+                                        processGroupId,
+                                        'controller-services',
+                                        response.id,
+                                        'edit'
+                                    ],
+                                    routeBoundary: response.postUpdateNavigationBoundary,
+                                    context: 'Controller Service'
+                                } as BackNavigation
+                            }
+                        });
```

Comment:
This condition needs to fire regardless of `response.postUpdateNavigationBoundary` existing.
The scenario that demonstrates this is editing a CS then clicking a referencing component. when it asks you to save before navigating, choose Yes. The CS gets saved, but this condition fails and the user is not routed to the referencing component.